source/agora/node/Validator.d

```diff
@@ -553,8 +557,27 @@ public class Validator : FullNode, API
         {
             log.trace("Sending Enrollment at height {} for {} cycles with {}",
                 block_height, this.params.ValidatorCycle, enroll_key);
-            this.network.peers.each!(p => p.client.sendEnrollment(
-                this.enroll_man.createEnrollment(enroll_key, block_height + 1)));
+            const enrollment = this.enroll_man.createEnrollment(enroll_key,
+                block_height + 1);
+
+            // Disable existing timer
+            if (emitter)
+                emitter.stop();
+
+            // Network tests use 1 second block interval, timeout should be
+            // adjusted accordingly
+            version (unittest) enum EnrollmentDelay = 10.msecs;
+            else enum EnrollmentDelay = 10.seconds;
```

Comment:
I think we still try to retrieve 1024 blocks at once, so that probably needs to be changed to otherwise 10secs might not be enough :)

Reply:
I just checked, nodes ask for 1024 blocks but they only return `MaxBatchBlocksSent` which is 20.